File: DSA Practice/LeetCode/153. Find Minimum in Rotated Sorted Array/solution.java
class Solution {
    public int findMin(int[] nums) {
        int m = Integer.MAX_VALUE;
        int left = 0;
        int right = nums.length - 1;

        while(left <= right){
            int cm = Math.min(nums[left], nums[right]);
            m = Math.min(m, cm);
            left++;
            right--;
        }

        return m;
    }
}

File: DSA Practice/LeetCode/153. Find Minimum in Rotated Sorted Array/solution.kt
class Solution {
    fun findMin(nums: IntArray): Int {
        var m = Integer.MAX_VALUE
        var left = 0
        var right = nums.size - 1

        while(left <= right){
            var cm = Math.min(nums[left], nums[right])
            m = Math.min(m, cm)
            left++
            right--
        }

        return m
    }
}